packages/parser/src/ethereum/index.ts

```diff
@@ -0,0 +1,180 @@
+          sendAddress,
+          receiveAddress,
+          sendValue.toString(10)
+        )
+      }
+
+      // network fee
+      const fees = new BigNumber(tx.fees ?? 0)
+      if (!fees.isNaN() && fees.gt(0)) {
+        pTx.fee = { caip19: 'ETH', value: fees.toString(10) }
+      }
+    }
+
+    if (address === receiveAddress) {
+      // receive amount
+      const receiveValue = new BigNumber(tx.value)
+      if (!receiveValue.isNaN() && receiveValue.gt(0)) {
+        pTx.transfers = this.aggregateTransfer(
+          pTx.transfers,
+          TransferType.Receive,
+          'ETH',
+          sendAddress,
+          receiveAddress,
+          receiveValue.toString(10)
+        )
+      }
+    }
+
+    tx.tokenTransfers?.forEach((transfer) => {
+      // FTX Token (FTT) name and symbol was set backwards on the ERC20 contract
+      if (transfer.token == '0x50D1c9771902476076eCFc8B2A83Ad6b9355a4c9') {
+        transfer.name = transfer.symbol
+        transfer.symbol = transfer.name
+      }
+
+      const token: Token = {
+        contract: transfer.token,
+        decimals: transfer.decimals,
+        name: transfer.name,
+      }
+
+      const transferArgs = [transfer.symbol, transfer.from, transfer.to, transfer.value, token] as const
+
+      // token send amount
+      if (address === transfer.from) {
+        pTx.transfers = this.aggregateTransfer(pTx.transfers, TransferType.Send, ...transferArgs)
+      }
+
+      // token receive amount
+      if (address === transfer.to) {
+        pTx.transfers = this.aggregateTransfer(pTx.transfers, TransferType.Receive, ...transferArgs)
+      }
+    })
+
+    internalTxs?.forEach((internalTx) => {
+      const transferArgs = ['ETH', internalTx.from, internalTx.to, internalTx.value] as const
+
+      // internal eth send
+      if (address === internalTx.from) {
+        pTx.transfers = this.aggregateTransfer(pTx.transfers, TransferType.Send, ...transferArgs)
+      }
+
+      // internal eth receive
+      if (address === internalTx.to) {
+        pTx.transfers = this.aggregateTransfer(pTx.transfers, TransferType.Receive, ...transferArgs)
+      }
+    })
+
+    return pTx
+  }
+  // keep track of all individual tx components and add up the total value transferred by to/from address
+  private aggregateTransfer(
+    transfers: Array<Transfer>,
+    type: TransferType,
+    caip19: string,
+    from: string,
+    to: string,
+    value: string,
+    token?: Token
+  ): Array<Transfer> {
+    const index = transfers?.findIndex((t) => t.type === type && t.caip19 === caip19 && t.from === from && t.to === to)
+    const transfer = transfers?.[index]
+
+    if (transfer) {
+      transfer.totalValue = new BigNumber(transfer.totalValue).plus(new BigNumber(value)).toString(10)
+      transfer.components.push({ value: value })
+      transfers[index] = transfer
+    } else {
+      transfers = [...transfers, { type, caip19, from, to, totalValue: value, components: [{ value: value }], token }]
```

Comment:
```suggestion
      return [...transfers, { type, caip19, from, to, totalValue: value, components: [{ value: value }], token }]
```

it's preferable to not reassign function argument values. I would expect eslint/typescript to complain about that.

Reply:
No complaints with strict tsconfig rules and default eslint. For the existing transfer case (first if statement) we reassign the value of the transfer, do you suggest a different way for that as well?